src/main/java/de/fraunhofer/aisec/cpg/passes/CallResolver.java

```diff
@@ -704,19 +1008,127 @@ private ConstructorDeclaration createConstructorDummy(
+    for (ConstructorDeclaration constructor : recordDeclaration.getConstructors()) {
+      if (constructor.hasSignature(signature)) {
+        return constructor;
+      }
+    }
+    return null;
+  }
+
+  private ConstructorDeclaration resolveConstructorWithDefaults(
+      ConstructExpression constructExpression,
+      List<Type> signature,
+      RecordDeclaration recordDeclaration) {
+    for (ConstructorDeclaration constructor : recordDeclaration.getConstructors()) {
+      if (!constructor.isImplicit() && signature.size() < constructor.getSignatureTypes().size()) {
+        List<Type> workingSignature =
+            getCallSignatureWithDefaults(constructExpression, constructor);
+        if (constructor.hasSignature(workingSignature)) {
+          for (Expression argument :
+              constructor
+                  .getDefaultParameters()
+                  .subList(signature.size(), constructor.getDefaultParameterSignature().size())) {
+            constructExpression.addArgument(argument, true);
+          }
+          return constructor;
+        }
+      }
+    }
+    return null;
+  }
+
+  private ConstructorDeclaration resolveConstructorWithImplicitCast(
+      ConstructExpression constructExpression, RecordDeclaration recordDeclaration) {
+    for (ConstructorDeclaration constructorDeclaration : recordDeclaration.getConstructors()) {
+      List<Type> workingSignature = new ArrayList<>(constructExpression.getSignature());
+      workingSignature.addAll(
+          constructorDeclaration
+              .getDefaultParameterSignature()
+              .subList(
+                  constructExpression.getArguments().size(),
+                  constructorDeclaration.getDefaultParameterSignature().size()));
+      if (compatibleSignatures(
+          constructExpression.getSignature(), constructorDeclaration.getSignatureTypes())) {
+        List<CastExpression> implicitCasts =
+            signatureWithImplicitCastTransformation(
+                constructExpression.getSignature(),
+                constructExpression.getArguments(),
+                constructorDeclaration.getSignatureTypes());
+        applyImplicitCastToArguments(constructExpression, implicitCasts);
+        return constructorDeclaration;
+      } else if (compatibleSignatures(
+          workingSignature, constructorDeclaration.getSignatureTypes())) {
+        List<CastExpression> implicitCasts =
+            signatureWithImplicitCastTransformation(
+                getCallSignatureWithDefaults(constructExpression, constructorDeclaration),
+                constructExpression.getArguments(),
+                constructorDeclaration.getSignatureTypes());
+        applyImplicitCastToArguments(constructExpression, implicitCasts);
+        addDefaultArgsToCall(constructorDeclaration, constructExpression);
+        return constructorDeclaration;
+      }
+    }
+    return null;
+  }
+
+  @NonNull
+  private ConstructorDeclaration getConstructorDeclarationCXX(
```

Comment:
Why does it have CXX in the name? It seems only a part of this is C++ specific. Can we add a short javadoc saying what this function is doing?